src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -344,31 +343,70 @@ public boolean randomBoolean()
         return ThreadLocalRandom.current().nextBoolean();
     }
 
+    public StringBuilder writeTsvHeaders()
+    {
+        StringBuilder builder = new StringBuilder();
+        List<String> fieldNames = new ArrayList<>(_columns.keySet());
+        fieldNames.removeAll(_autoGeneratedFields);
+
+        builder.append(String.join("\t", fieldNames));
+        builder.append("\n");
+
+        return builder;
+    }
+
+    public String rowToString(List<String> fieldNames, Map<String, Object> row)
+    {
+        StringBuilder builder = new StringBuilder();
+        List<String> values = new ArrayList<>();
+        for (String name : fieldNames)
+        {
+            Object value = row.get(name);
+            values.add(value != null ? String.valueOf(value) : "");
+        }
+        builder.append(String.join("\t", values));
+        builder.append("\n");
+
+        return builder.toString();
+    }
+
     /**
      * generates tsv-formatted content using the rows in the current instance;
      * @return TSV formatted representation of generated rows
      */
     public String writeTsvContents()
     {
-        StringBuilder builder = new StringBuilder();
         List<String> fieldNames = new ArrayList<>(_columns.keySet());
         fieldNames.removeAll(_autoGeneratedFields);
-
-        builder.append(String.join("\t", fieldNames));
-        builder.append("\n");
+        StringBuilder builder = writeTsvHeaders();
 
         for (Map<String, Object> row : _rows)
         {
-            List<String> values = new ArrayList<>();
-            for (String name : fieldNames)
+            builder.append(rowToString(fieldNames, row));
+        }
+        return builder.toString();
+    }
+
+    public File writeGeneratedDataToFile(int numberOfRowsToGenerate, String fileName) throws IOException
+    {
+        String headers = writeTsvHeaders().toString();
+        File file = new File(TestFileUtils.getTestTempDir(), fileName);
+        FileUtils.forceMkdirParent(file);
+
+        try(FileOutputStream stream = new FileOutputStream(file))
```

Comment:
Should probably use a PrintWriter for this. I don't know the particulars exactly but I think it buffers more efficiently and uses the correct file encoding.
```suggestion
        try (PrintWriter stream = PrintWriters.getPrintWriter(file))
```

Reply:
Thanks.
PrintWriter doesn't require me to feed bytes into the stream, it takes line-by-line values as String.